### ISSUE TITLE
The problem of using self-built dataset to train models and test self-built dataset

### ISSUE THREAD

**Issue by mjw123bs** [hlwang1124/SNE-RoadSeg#27]
Hi, I am a newbie in semantic segmentation. Can the images marked with labelme be used as a dataset? If so, how to create a dataset and how to use it, do I need to retrain the model with a self-built dataset? How can the trained model be used for testing?
I wait for your response.

**Comment by hlwang1124** [comment]
Hi, @mjw123bs .
1. We use the label images that have the following format: 0 for the unlabeled and 1 for the freespace. `labelme` can generate such labels.
2. If you want to use your own dataset, just prepare all the data including RGB, depth and label images, and split them into 3 sets as mentioned in `README`. You also need to revise the camera intrinsic parameters in the dataloader. Then, you can use the provided training and testing scripts.
3. You can try the provided pre-trained weights. However, if your domain has much difference from the KITTI dataset, I am afraid that the performance will not be good.

**Comment by mjw123bs** [comment]
I'm glad you can reply me quickly. I still have some questions. Should the json file generated by labelme correspond to the images in gt_images? What do the files in the calib folder do? Is it not necessary to make my own dataset?

**Comment by hlwang1124** [comment]
1. I remember that `labelme` can transform the json files to images, and we use the image-format ground truth.
2. `calib` provides the camera intrinsic parameters. We use it for the KITTI dataset, since KITTI provides a calibration file for each image. If you use your own dataset and the camera intrinsic parameters are the same for all images, you can omit `calib` and just fix the camera intrinsic parameters in the dataloader.

**Comment by mjw123bs** [comment]
### calib

**Comment by mjw123bs** [comment]
Okay,Thank you for your reply, @hlwang1124.
then which one is dataloader under the **data**? How to modify the camera intrinsic parameters in the dataloader? Are the camera intrinsic parameters the road edge parameters of the labelme?

**Comment by mjw123bs** [comment]
Okay,Thank you for your reply, @hlwang1124.

then which one is dataloader under the `data`  ?How to modify the camera intrinsic parameters in the dataloader? Are the camera intrinsic parameters the road edge parameters of the labelme?

**Comment by mjw123bs** [comment]
> 
> 
> Okay,Thank you for your reply, @hlwang1124.
> 
> then which one is dataloader under the `data` ?How to modify the camera intrinsic parameters in the dataloader? Are the camera intrinsic parameters the road edge parameters of the labelme?

dataloader is `kitti_dataset.py`, right?

**Comment by mjw123bs** [comment]
> 
> 
>     1. I remember that `labelme` can transform the json files to images, and we use the image-format ground truth.
> 
>     2. `calib` provides the camera intrinsic parameters. We use it for the KITTI dataset, since KITTI provides a calibration file for each image. If you use your own dataset and the camera intrinsic parameters are the same for all images, you can omit `calib` and just fix the camera intrinsic parameters in the dataloader.

` If you use your own dataset and the camera intrinsic parameters are the same for all images, you can omit `calib` and just fix the camera intrinsic parameters in the dataloader.`What does it mean？No need for `calib` folder？

**Comment by hlwang1124** [comment]
1. Yes, you can implement your own dataloader based on `kitti_dataset.py`.
2. Our SNE requires the camera intrinsic parameter to compute surface normal. The camera intrinsic parameter is an inherent parameter for each camera. It can be obtained by calibration, and I attach a link [here](http://ksimek.github.io/2013/08/13/intrinsic/) for your reference. If your dataset is collected by only one camera, the camera intrinsic parameter can be regarded as the same for all images. Therefore, you do not require `calib` folder to store different parameters for different images. Instead, you can only write a fixed one in the dataloader. Specifically, just write your camera intrinsic parameters as a torch tensor to replace Line110-111 in `kitti_dataset.py`.

**Comment by mjw123bs** [comment]
> 
> 
>     1. Yes, you can implement your own dataloader based on `kitti_dataset.py`.
> 
>     2. Our SNE requires the camera intrinsic parameter to compute surface normal. The camera intrinsic parameter is an inherent parameter for each camera. It can be obtained by calibration, and I attach a link [here](http://ksimek.github.io/2013/08/13/intrinsic/) for your reference. If your dataset is collected by only one camera, the camera intrinsic parameter can be regarded as the same for all images. Therefore, you do not require `calib` folder to store different parameters for different images. Instead, you can only write a fixed one in the dataloader. Specifically, just write your camera intrinsic parameters as a torch tensor to replace Line110-111 in `kitti_dataset.py`.

Hi,Thank you a lot for your reply! Your suggestions are valuable. I am going to try as you said. I have other small questions. Does the image format have to be in png format? What program did you use to generate the pictures in gt_image_2 as following?

**Comment by hlwang1124** [comment]
You can use your own ground truth format. Just implement your own dataloader to enable the ground truth tensor has the following format: 0 for the unlabeled and 1 for the freespace.

**Comment by mjw123bs** [comment]
Okay, I will do what you said first, and I will contact you if I encounter other problems, Thank you again！
